### DISCUSSION TITLE
Please Help with optimizing config

### DISCUSSION THREAD

**Discussion by JackHsuan** [swar/Swar-Chia-Plot-Manager#526] [Q&A] [answered]
Equipments:
SSD : M.2 NVMe SSD 2T *2 (Un-popular Brand) RAID0 IN WIN10
Mem : 64G (32g 3600Ghz *2)
Cpu: AMD Ryzen 9 5900X (12C 24T)
motherboard: ASUS B550-XE

I have no idea about how to optimize the plotting,I use Chia.exe plot 10*32 each time.
I hope someone has recommend configuration.

Thank you.
Best Regards.

**Comment by Lyushen** [comment, marked as answer]
> SSD : M.2 NVMe SSD 2T *2 (Un-popular Brand) RAID0 IN WIN10

What does it means? Did u raid it through the bios or just through device manager?
I recommend you separate them and run different tasks for everyone, windows easily bottlenecking IO, so run it separately is reasonable instead of raid.
Try to create 2 tasks (one per nvme) with 4 cpu threads, stagger time min 60 mins, run 3 on phase one per task (6 will be in general, but I recommend only 2*2) and 6 max_concurrent per task, something like it 
```
manager:
check_interval: 60
log_level: ERROR

global:
max_concurrent: 12

jobs:
- name: NVMe 1
max_plots: 999
farmer_public_key:
pool_public_key:
temporary_directory: D:\
temporary2_directory:
destination_directory: G:\
size: 32
bitfield: true
threads: 3
buckets: 128
memory_buffer: 4000
max_concurrent: 6
max_concurrent_with_start_early: 6
stagger_minutes: 60
max_for_phase_1: 3
concurrency_start_early_phase: 4
concurrency_start_early_phase_delay: 30
temporary2_destination_sync: false

- name: NVMe 2
max_plots: 999
farmer_public_key:
pool_public_key:
temporary_directory: E:\
temporary2_directory:
destination_directory: G:\
size: 32
bitfield: true
threads: 3
buckets: 128
memory_buffer: 4000
max_concurrent: 6
max_concurrent_with_start_early: 6
stagger_minutes: 60
max_for_phase_1: 3
concurrency_start_early_phase: 4
concurrency_start_early_phase_delay: 30
temporary2_destination_sync: false
```
If you do only 2*2 max_for_phase_1 one you can left some process resources for other phases, 4 threads *3 per phase one = 12 threads per task or 24 within 2 tasks actually can bottleneck your other phases. So you can do 3 threads and 3 per phase one or 4 threads and 2 max phase one per task. Play with it but calculate smart your process power, it's still AMD. I'm running 10850k on 5ghz and still can't handle more than 5 per phase one in total. The RAM doesn't matter, and the rest as well, only this two parameters you should pay attention

**Reply by JackHsuan** [reply]
Hi Lyushen,

appreciate for your detailed explaination.
I did the Raid0 through the bios not by windows, Is it better to use one task or better to do it separately?
and I am confusing about the toddsby's configuration, it use 18 threads, can i do it like this?

Thanks,
Best Regards.
Jackhsuan

**Reply by Lyushen** [reply]
Separatly using most of IOPS, because it limited by windows and the way it works, it's not actually speeding up the process at all.
No no no, only 2-4 threads per task. The program is super poorly coded and it cannot use potential of multithreading, actually adding more than 5 threads doing opposite, slowing down the process. In exclusive some cases 5 threads per task, but what I've tested I have no differece after 3 threads per task because I run 10 tasks in general, so there is no free cpu power that could be utilized with my 10core@5GHz processor. But AMD is another case, they could benefit from 5 threads per task. You should figure it out not by just time, use logs and monitor every stage, how much it uses, if less than 150% on phase1 after checkpoint, it could be because of bottleneck cpu or if cpu is have resources, than it could be NVMe. Actually a lot of people coming here and saying the performance issue and flashing their raid 0. I believe you suffering from 2x drop performance in comparison when you do 2 diff tasks. If your cpu is free and it not hits even 75% it should be raid. Try it mate. Gl tho

**Reply by JackHsuan** [reply]
Ok, I got it. Thank you so much for your great help.
